src/haxesharp/collections/Linq.hx

```diff
@@ -58,6 +59,60 @@ class Linq<T>
     }
 
     /**
+    If a predicate is supplied, returns the only element to satisfy the predicate.  Throws an exception if more than one such element exists.
+    If a predicate is not supplied, return the only element in the array.  Throws an exception if the array contains more than one element.
+    In both cases, and exception is thrown if the provided array is null.
+    */
+    public static function single<T>(array:Array<T>, ?predicate:T->Bool):T
+    {
+        if (array == null)
+        {
+            throw new ArgumentNullException('"array" argument cannot be null.');

```

Comment:
I think you can remove this case. I don't think it's possible to get a null array on an extension method. You would have to invoke something like `var x:Array<T> = null; x.single`. If that's possible. ugh, we have to add this to the other `Linq` methods too! (I forgot about this case.)

Reply:
Yeah, assuming the same semantics apply to extension methods in Haxe as in C#, which in this case it seems they do, the object from which the extension method is called could be null.